File: server/app/models/problems.py
from enum import StrEnum

from pydantic import BaseModel


class Difficulty(StrEnum):
    EASY = "easy"
    MEDIUM = "medium"
    HARD = "hard"


class TopicTag(StrEnum):
    ARRAY = "array"
    BACKTRACKING = "backtracking"


class ProblemOptions(BaseModel):
    difficulty: Difficulty
    topic_tag: TopicTag


class Problem(BaseModel):
    pass

File: frontend/types/problems.ts
import { z } from 'zod';

export const difficulty = z.enum(['EASY', 'MEDIUM', 'HARD']);
export type Difficulty = z.infer<typeof difficulty>;

export const problemOptions = z.object({
  difficulty: difficulty,
});
export type ProblemOptions = z.infer<typeof problemOptions>;

export const defaultProblemOptions: ProblemOptions = {
  difficulty: 'MEDIUM',
};
